### DISCUSSION TITLE
Firebase or Gloud Identity Custom SMTP not working

### DISCUSSION THREAD

**Discussion by Chrissi2812** [postalserver/postal#2938] [Help with using Postal] [answered]
Has anyone an Idea how to make google work with the Postal as SMTP?

I have setup the server in Firebase as described under "Credentials -> Read more about sending outgoing e-mails".
According to the Logs it connects fine but Google drops the connection immediately.

```
smtp_1    | 2024-04-19 09:43:25 +0000 DEBUG  Client identified as 209.85.161.72 trace_id=ZKJ58NBX component=smtp-server
smtp_1    | 2024-04-19 09:43:25 +0000 DEBUG  <= EHLO mail-oo1-f72.google.com trace_id=ZKJ58NBX component=smtp-server
smtp_1    | 2024-04-19 09:43:25 +0000 DEBUG  => 250-My capabilities are trace_id=ZKJ58NBX component=smtp-server
smtp_1    | 2024-04-19 09:43:25 +0000 DEBUG  => 250-STARTTLS trace_id=ZKJ58NBX component=smtp-server
smtp_1    | 2024-04-19 09:43:25 +0000 DEBUG  => 250 AUTH CRAM-MD5 PLAIN LOGIN trace_id=ZKJ58NBX component=smtp-server
smtp_1    | 2024-04-19 09:43:25 +0000 DEBUG  <= STARTTLS trace_id=ZKJ58NBX component=smtp-server
smtp_1    | 2024-04-19 09:43:25 +0000 DEBUG  => 220 Ready to start TLS trace_id=ZKJ58NBX component=smtp-server
smtp_1    | 2024-04-19 09:43:26 +0000 DEBUG  Connection closed trace_id=ZKJ58NBX component=smtp-server
```

I had another custom SMTP setup working before and the Responses from the servers are quite different

Working Custom SMTP:
```
220 REDACTED.webpack.hosteurope.de ESMTP Host Europe Mail Service Fri, 19 Apr 2024 11:56:11 +0200
EHLO mail-oo1-f72.google.com
250-REDACTED.webpack.hosteurope.de Hello REDACTED [REDACTED]
250-SIZE 52428800
250-8BITMIME
250-PIPELINING
250-AUTH PLAIN LOGIN
250-STARTTLS
250 HELP
```

Postal Custom SMTP:
```
220 postal.REDACTED.com ESMTP Postal/CYOC5NO0
EHLO mail-oo1-f72.google.com
250-My capabilities are
250-STARTTLS
250 AUTH CRAM-MD5 PLAIN LOGIN
```

I also verified via `curl` that my postal can send E-Mails via SMTP (Tried all AUTH, also with and without forced ssl).

**Comment by willpower232** [comment]
Are you able to turn off the TLS in firebase and see what its logs look like then?

**Reply by Chrissi2812** [reply]
No Firebase aswell as the Gcloud Interface only let me toggle between STARTTLS & SSL.
And GCloud only logs that it trying to reset password and sending the Mail via the Custom SMTP but not error messages for the SMTP part are logged...

**Reply by willpower232** [reply]
You could try watching the Postal logs when Firebase tries to send a message in case there is an exception or other error that Postal isn't outputting to them.

**Reply by Chrissi2812** [reply]
Thanks for your help, I watched the logs with `postal logs --follow` and didn't saw any exceptions.

Therefore I contacted the firebase support about this issue. Because my postal instance with all other test methods I tried worked without any trouble.
My guess is that firebase has some trouble with the `/` or the `-` in the username. But that's pure speculation. 

I will post an update as soon as the firebase support has either resolved it or provided me with information what is wrong.

**Comment by Chrissi2812** [comment, marked as answer]
Okay the Problem is now solved. 
I used only the leave certificate and not the fullchain that Let's Encrypt produced.
After fixing that Firebase is now able to send without any issues via Postal.
